src/sources.cpp

```diff
@@ -249,12 +249,13 @@ static bool push_files(std::vector<std::string> &out, const std::string &path, i
   auto dir = fdopendir(dirfd);
   if (!dir) {
     close(dirfd);
+    fprintf(stderr, "Failed to fdopendirt %s: %s\n", path.c_str(), strerror(errno));
     return true;
   }
 
   struct dirent *f;
   bool failed = false;
-  for (errno = 0; !failed && (f = readdir(dir)); errno = 0) {
+  for (errno = 0; 0 != (f = readdir(dir)); errno = 0) {
```

Comment:
When I look at `0 != (f = readdir(dir))`, I start thinking this could be a while loop instead

Reply:
Well, the `0 != ` part is necessary to prevent a warning.

I guess it might be more readable as a while loop, though?:
```
errno = 0;
while (0 != (f = readdir(dir)) {
  ...
  errno = 0;
}
```
... maybe not.

Reply:
```
while (true) {
  errno = 0
  f = readdir(dir);
  if (f != 0)
    break;
  
  // other stuff
}
```

Reply:
It's not really important for this PR though, it's more a style-opinion

Reply:
I dunno. I think using 'break' instead of a proper loop guard is actually worse.
The syntax of `if (0 != (x = foo()))` is pretty common in C++.

Reply:
I prefer to functions exit at the last line with a single return and have loops eschew break/continue.

Only if there is a significant reduction in overall complexity do I violate that.